### PR TITLE
ci(kichen+travis): test with pre-salted Docker images

### PR DESCRIPTION
**Pull request by n-rodriguez**

### CODE REVIEW

**Comment by n-rodriguez** [comment]
Fixed: https://travis-ci.com/n-rodriguez/systemd-formula/jobs/199847337
But OpenSuse failed : https://travis-ci.com/n-rodriguez/systemd-formula/builds/111655701

**Comment by myii** [comment]
It's only one of them (`opensuse-leap-15-2019-2-py3`) and only failing a few of the tests:

```kitchen
Profile: Systemd Formula (systemd)
Version: (not specified)
Target:  ssh://kitchen@localhost:32768
  ✔  Systemd package: should be installed
     ✔  System Package systemd should be installed
  ×  Systemd units: should match desired lines (1 failed)
     ✔  File /etc/systemd/system/syncthing-someuser.service type should eq :file
     ✔  File /etc/systemd/system/syncthing-someuser.service mode should cmp == "0644"
     ✔  File /etc/systemd/system/syncthing-someuser.service owner should eq "root"
     ✔  File /etc/systemd/system/syncthing-someuser.service group should eq "root"
     ×  Service syncthing-someuser.service should be enabled
     expected that `Service syncthing-someuser.service` is enabled
  ×  Systemd Networkd: should match desired lines (3 failed)
     ✔  File /etc/systemd/network type should eq :directory
     ✔  File /etc/systemd/network mode should cmp == "0755"
     ✔  File /etc/systemd/network owner should eq "root"
     ✔  File /etc/systemd/network group should eq "root"
     ✔  File /etc/systemd/network/99-default.link type should eq :file
     ✔  File /etc/systemd/network/99-default.link mode should cmp == "0644"
     ✔  File /etc/systemd/network/99-default.link owner should eq "root"
     ✔  File /etc/systemd/network/99-default.link group should eq "root"
     ✔  File /etc/systemd/network/eth0.network type should eq :file
     ✔  File /etc/systemd/network/eth0.network mode should cmp == "0644"
     ✔  File /etc/systemd/network/eth0.network owner should eq "root"
     ✔  File /etc/systemd/network/eth0.network group should eq "root"
     ✔  File /etc/systemd/network/br0.netdev type should eq :file
     ✔  File /etc/systemd/network/br0.netdev mode should cmp == "0644"
     ✔  File /etc/systemd/network/br0.netdev owner should eq "root"
     ✔  File /etc/systemd/network/br0.netdev group should eq "root"
     ✔  File /etc/systemd/network/10-dmz.link type should eq :file
     ✔  File /etc/systemd/network/10-dmz.link mode should cmp == "0644"
     ✔  File /etc/systemd/network/10-dmz.link owner should eq "root"
     ✔  File /etc/systemd/network/10-dmz.link group should eq "root"
     ✔  File /etc/systemd/network/10-internet.link type should eq :file
     ✔  File /etc/systemd/network/10-internet.link mode should cmp == "0644"
     ✔  File /etc/systemd/network/10-internet.link owner should eq "root"
     ✔  File /etc/systemd/network/10-internet.link group should eq "root"
     ×  Service systemd-networkd should be enabled
     expected that `Service systemd-networkd` is enabled
     ×  Service systemd-networkd should be running
     expected that `Service systemd-networkd` is running
     ×  Service systemd-networkd-wait-online should be enabled
     expected that `Service systemd-networkd-wait-online` is enabled
Profile Summary: 1 successful control, 2 control failures, 0 controls skipped
Test Summary: 29 successful, 4 failures, 0 skipped
```

**Comment by aboe76** [comment]
@n-rodriguez can't see why this would fail only on opensuse 15.0,
started the same container on my own test machine:
```
docker run --privileged --rm -it netmanagers/salt-2019.2-py3:opensuse-leap-15 /usr/lib/systemd/systemd
```
next entered the container:
```
docker exec -it 8cfef52997ad bash
```
```
cfef52997ad:/etc/systemd/network # systemctl enable systemd-networkd --now
Created symlink /etc/systemd/system/network.service → /usr/lib/systemd/system/systemd-networkd.service.
Created symlink /etc/systemd/system/dbus-org.freedesktop.network1.service → /usr/lib/systemd/system/systemd-networkd.service.
Created symlink /etc/systemd/system/multi-user.target.wants/systemd-networkd.service → /usr/lib/systemd/system/systemd-networkd.service.
Created symlink /etc/systemd/system/sockets.target.wants/systemd-networkd.socket → /usr/lib/systemd/system/systemd-networkd.socket.
Created symlink /etc/systemd/system/network-online.target.wants/systemd-networkd-wait-online.service → /usr/lib/systemd/system/systemd-networkd-wait-online.service.
8cfef52997ad:/etc/systemd/network # systemctl status systemd-networkd
● systemd-networkd.service - Network Service
   Loaded: loaded (/usr/lib/systemd/system/systemd-networkd.service; enabled; vendor preset: disabled)
   Active: active (running) since Tue 2019-05-14 17:33:30 UTC; 3s ago
     Docs: man:systemd-networkd.service(8)
 Main PID: 345 (systemd-network)
   Status: "Processing requests..."
    Tasks: 1 (limit: 4915)
   CGroup: /docker/8cfef52997adc8d3f05498f4e725ab53130d39fc6fee410310dee08343a9d9b8/system.slice/systemd-networkd.service
           └─345 /usr/lib/systemd/systemd-networkd

May 14 17:33:30 8cfef52997ad systemd[1]: Starting Network Service...
May 14 17:33:30 8cfef52997ad systemd-networkd[345]: Enumeration completed
May 14 17:33:30 8cfef52997ad systemd[1]: Started Network Service.
```
so it should work...

**Comment by n-rodriguez** [comment]
> I have tested the formula on tumbleweed(opensuse rolling release) so I'm ok with one test failing...

it might be a bug within inspec...

**Comment by n-rodriguez** [comment]
> it might be a bug within inspec...

no... systemd-network was not enabled :

```
cfef52997ad:/etc/systemd/network # systemctl enable systemd-networkd --now
Created symlink /etc/systemd/system/network.service → /usr/lib/systemd/system/systemd-networkd.service.
Created symlink /etc/systemd/system/dbus-org.freedesktop.network1.service → /usr/lib/systemd/system/systemd-networkd.service.
Created symlink /etc/systemd/system/multi-user.target.wants/systemd-networkd.service → /usr/lib/systemd/system/systemd-networkd.service.
Created symlink /etc/systemd/system/sockets.target.wants/systemd-networkd.socket → /usr/lib/systemd/system/systemd-networkd.socket.
Created symlink /etc/systemd/system/network-online.target.wants/systemd-networkd-wait-online.service → /usr/lib/systemd/system/systemd-networkd-wait-online.service
```

**Comment by javierbertoli** [comment]
@n-rodriguez the issue is an `inspec/train`+`suse/opensuse` one, we hit the same problem on the `template-formula`, see [here](https://github.com/saltstack-formulas/template-formula/blob/master/test/integration/default/controls/services_spec.rb#L7-L10)

**Comment by aboe76** [comment]
@javierbertoli thanks for finding this.

**Comment by n-rodriguez** [comment]
well... that's weird...

```sh
nicolas@cloud:~/SALT/ufw-formula$ bundle exec kitchen login default-opensuse-leap-15-2019-2-py3
Last login: Wed May 15 19:17:33 2019 from 172.17.0.1
Have a lot of fun...
kitchen@ab3c49a5edd3:~> cat /etc/SuSE-release 
NAME="openSUSE Leap"
VERSION="15.0"
ID="opensuse-leap"
ID_LIKE="suse opensuse"
VERSION_ID="15.0"
PRETTY_NAME="openSUSE Leap 15.0"
ANSI_COLOR="0;32"
CPE_NAME="cpe:/o:opensuse:leap:15.0"
BUG_REPORT_URL="https://bugs.opensuse.org"
HOME_URL="https://www.opensuse.org/"
kitchen@ab3c49a5edd3:~> 
```

**Comment by n-rodriguez** [comment]
nicolas@cloud:~/SALT/systemd-formula$ bundle exec kitchen verify default-opensuse-leap-15-2019-2-py3
-----> Starting Kitchen (v2.2.4)
-----> Verifying <default-opensuse-leap-15-2019-2-py3>...
[2019-05-15T21:38:59+02:00] WARN: DEPRECATION: InSpec Attributes are being renamed to InSpec Inputs to avoid confusion with Chef Attributes. Use --input-file on the command line instead of --attrs.
       Loaded systemd 
[2019-05-15T21:39:01+02:00] WARN: DEPRECATION: The service `be_running?` matcher is deprecated. This is only allowed for compatibility with ServerSpec (used at /data/nicolas/SALT/systemd-formula/test/integration/default/controls/networkd_spec.rb:48)
e
Profile: Systemd Formula (systemd)
Version: (not specified)
Target:  ssh://kitchen@localhost:32778

  ✔  Systemd units: should match desired lines
     ✔  File /etc/systemd/system/syncthing-someuser.service type should eq :file
     ✔  File /etc/systemd/system/syncthing-someuser.service mode should cmp == "0644"
     ✔  File /etc/systemd/system/syncthing-someuser.service owner should eq "root"
     ✔  File /etc/systemd/system/syncthing-someuser.service group should eq "root"
     ✔  Service syncthing-someuser.service should be enabled
  ✔  Systemd Networkd: should match desired lines
     ✔  File /etc/systemd/network type should eq :directory
     ✔  File /etc/systemd/network mode should cmp == "0755"
     ✔  File /etc/systemd/network owner should eq "root"
     ✔  File /etc/systemd/network group should eq "root"
     ✔  File /etc/systemd/network/99-default.link type should eq :file
     ✔  File /etc/systemd/network/99-default.link mode should cmp == "0644"
     ✔  File /etc/systemd/network/99-default.link owner should eq "root"
     ✔  File /etc/systemd/network/99-default.link group should eq "root"
     ✔  File /etc/systemd/network/eth0.network type should eq :file
     ✔  File /etc/systemd/network/eth0.network mode should cmp == "0644"
     ✔  File /etc/systemd/network/eth0.network owner should eq "root"
     ✔  File /etc/systemd/network/eth0.network group should eq "root"
     ✔  File /etc/systemd/network/br0.netdev type should eq :file
     ✔  File /etc/systemd/network/br0.netdev mode should cmp == "0644"
     ✔  File /etc/systemd/network/br0.netdev owner should eq "root"
     ✔  File /etc/systemd/network/br0.netdev group should eq "root"
     ✔  File /etc/systemd/network/10-dmz.link type should eq :file
     ✔  File /etc/systemd/network/10-dmz.link mode should cmp == "0644"
     ✔  File /etc/systemd/network/10-dmz.link owner should eq "root"
     ✔  File /etc/systemd/network/10-dmz.link group should eq "root"
     ✔  File /etc/systemd/network/10-internet.link type should eq :file
     ✔  File /etc/systemd/network/10-internet.link mode should cmp == "0644"
     ✔  File /etc/systemd/network/10-internet.link owner should eq "root"
     ✔  File /etc/systemd/network/10-internet.link group should eq "root"
     ✔  Service systemd-networkd should be enabled
     ✔  Service systemd-networkd should be running
     ✔  Service systemd-networkd-wait-online should be enabled
  ✔  Systemd package: should be installed
     ✔  System Package systemd should be installed


Profile Summary: 3 successful controls, 0 control failures, 0 controls skipped
Test Summary: 33 successful, 0 failures, 0 skipped
       Finished verifying <default-opensuse-leap-15-2019-2-py3> (0m3.54s).
-----> Kitchen is finished. (0m5.27s)
nicolas@cloud:~/SALT/systemd-formula$

**Comment by n-rodriguez** [comment]
@javierbertoli  https://github.com/inspec/train/pull/451

**Comment by myii** [comment]
Great work @n-rodriguez!

**Comment by aboe76** [comment]
@n-rodriguez nice find

**Comment by aboe76** [comment]
@n-rodriguez and @myii can we merge this one?

File: .travis.yml [hunk 1]
```diff
@@ -11,15 +11,27 @@ language: ruby
...
+    - INSTANCE: default-centos-7-2018-3-py2
+    - INSTANCE: default-fedora-29-2018-3-py2
+    # TODO: Use this when fixed instead of `opensuse-leap-42`
+    # Ref: https://github.com/netmanagers/salt-image-builder/issues/2
+    # - INSTANCE: default-opensuse-leap-15-2018-3-py2
+    - INSTANCE: default-opensuse-leap-42-2018-3-py2
+    - INSTANCE: default-debian-8-2017-7-py2
+    - INSTANCE: default-ubuntu-1604-2017-7-py2
+    - INSTANCE: default-fedora-28-2017-7-py2
+    - INSTANCE: default-opensuse-leap-42-2017-7-py2
```
Comment (myii) [inline .travis.yml]: Please use the reduced matrix merged into the `template-formula` recently, or similar.

**Comment by n-rodriguez** [comment]
> @n-rodriguez and @myii can we merge this one?

I'd like to, but I'm waiting for release 2.1.6 of train on rubygems.

**Comment by myii** [comment]
@n-rodriguez Merged, thanks for this PR and for fixing `train` upstream!

**Pull request merged**